### DISCUSSION TITLE
authlib-injector v1.2.0 升级常见问题解答 FAQ

### DISCUSSION THREAD

**Discussion by yushijinhun** [yushijinhun/authlib-injector#174] [General]
本 Discussion 主要提供 authlib-injector v1.2.0 升级过程中的常见问题解答（FAQ），如您认为此列表中的解答需要改进，或者希望增加新的问题，可以在此处讨论。

------

**Q: 我在进入服务器时遇到错误 `无效的个人信息公钥签名，请尝试重启游戏`，该怎么办？**

A: 请确保您的客户端和服务端均使用 authlib-injector v1.2.0 及以上版本。对于客户端而言，启动器（HMCL、BakaXL、PCL）会自动更新 authlib-injector，因此其一般都是最新的；对于服务器而言，则需要服主升级 authlib-injector。

------

**Q: 如何升级服务端的 authlib-injector？**

A: 首先，请在 [authlib-injector.yushi.moe](https://authlib-injector.yushi.moe/) 下载最新的 authlib-injector，然后替换掉旧版的 authlib-injector 的 JAR 文件。随后，如果您使用 MC 1.19+，请修改 `server.properties` 中的 `enforce-secure-profile`，将其设置为 `true`，这将消除掉进入服务器时的 `无法验证聊天消息` 弹窗。

------

**Q: 如何升级 BungeeCord / Velocity 的 authlib-injector？**

A: 请确保使用 **v1.2.1 及以上版本**的 authlib-injector。对于 BungeeCord，请开启 `config.yml` 中的 `enforce_secure_profile`；对于 Velocity，请开启 `velocity.toml` 中的 `force-key-authentication`。

------

**Q: 我的服务端使用 MultiLogin，而不是 authlib-injector，该怎么办？**

A: 目前 MultiLogin 尚不兼容 authlib-injector v1.2.0+，该问题已在 https://github.com/CaaMoe/MultiLogin/issues/56 追踪，请等待 MultiLogin 发布新版本。

> :point_right: **目前的暂时解决方法（由 MultiLogin 作者提供）** :point_left:
>
> 在服务端使用 MultiLogin 的同时，安装 authlib-injector。（该方法几乎是唯一的解决方法）
>
> 特别声明: 这两个程序装一起是不会有冲突和副作用的，MultiLogin 会将 authlib-injector 的登录验证功能失效，仅保留其他的优化功能。但是需要提供一个有效的 Yggdrasil api root 地址提供给 authlib-injector，请放心使用。
>
> （目前v3和v5的velocity已修复这个异常，需要自己编译。）
>
> （服务器内发言仍然有些问题，请谨慎聊天。必要时可以通过指令为自己的正版账号添加多一种外置登录方式）

------

**Q: 服务端已经升级到了 authlib-injector v1.2.0+，但玩家依然无法进服该怎么办？**

A: 请确保客户端使用的 authlib-injector 也是 v1.2.0+，方法见下一条解答。

------

**Q: 如何知道客户端使用的 authlib-injector 版本？**

A: 请打开客户端 `.minecraft` 目录下的 `authlib-injector.log` 日志文件，其中有 authlib-injector 的版本号。

------

**Q: 我的启动器没有更新最新版本 (v1.2.0+) 的 authlib-injector，该怎么办？**

A: 请尝试更换下载源并重启启动器。如果问题依旧，请联系启动器作者，或通过文末的方式联系本项目开发者寻求帮助。

------

**Q: 我在服务端使用 Geyser，基岩版用户进服时提示 `无效的个人信息公钥签名，请尝试重启游戏`，该怎么办？**

A: 这是 Geyser 的 bug，目前已由 https://github.com/GeyserMC/Geyser/issues/3187 追踪。您可以暂时将 `server.properties` 中的 `enforce-secure-profile` 设置为 `false`，作为 workaround，代价是 Java 版玩家进服时会有 `无法验证聊天消息` 弹窗。

------

**Q: 这一更新是否影响 MC 1.19+ 以外的版本？**

A: authlib-injector v1.2.0 更新只影响 MC 1.19+，对低于 1.19 的版本不应存在任何影响，不会遇到本文中所述的问题。

------

**Q: 本次更新的内容是什么？**

A: 更新内容针对 MC 1.19+ 版本，主要为两点：1) 服务器可以在 `server.properties` 中开启 `enforce-secure-profile`，这样玩家进服时就不会遇到 `无法验证聊天消息` 的弹窗；2) 玩家在服务器内的聊天消息不会再有无法验证签名的感叹号标志。

------

**Q: 更新后，玩家进服时依然会有 `无法验证聊天消息` 弹窗，该怎么办？**

A: 请修改 `server.properties` 中的 `enforce-secure-profile`，将其设置为 `true`。

------

**Q: 我的问题没有出现在上面的 FAQ 中，该怎么办？**

A: 请加入 Telegram 群组 [@authlib_injector](https://t.me/authlib_injector)，或 QQ 群 926979364 提问。您也可以在[本项目的 Discussions 提问](https://github.com/yushijinhun/authlib-injector/discussions/new?category=q-a)，或[提交一个 Issue](https://github.com/yushijinhun/authlib-injector/issues/new)。

**Comment by Ha22yX** [comment]
我使用了此插件后，littleskin的账号可以登入，但是mc正版账号无法登入了，怎么办

**Reply by yeying-xingchen** [reply]
这是正常的，如有需要，请使用MultiLogin。
